panels/models/attraction.py

```diff
@@ -33,6 +34,159 @@ def sluggify(s):
     return RE_SLUG.sub('-', s).lower().strip('-')
 
 
+def groupify(items, keys, val_key=None):
+    """
+    Groups a list of items into nested OrderedDicts based on the given keys.
```

Comment:
Whoa! With this kind of flexibility, this could be its own library! Thank you for the docs!